sovereign-forge/src/common/sc_deploy.rs

```diff
@@ -1,67 +1,118 @@
 use crate::err_msg;
 use multiversx_sc::{
     imports::OptionalValue,
-    types::{MultiValueEncoded, ReturnsResult},
+    sc_panic,
+    types::{ManagedAsyncCallResult, MultiValueEncoded},
 };
 use proxies::chain_factory_proxy::ChainFactoryContractProxy;
 use structs::{
     configs::{EsdtSafeConfig, SovereignConfig},
     fee::FeeStruct,
-    forge::ContractInfo,
+    forge::{ContractInfo, ScArray},
 };
 
 #[multiversx_sc::module]
 pub trait ScDeployModule: super::utils::UtilsModule + super::storage::StorageModule {
     #[inline]
     fn deploy_chain_config(
         &self,
+        chain_id: &ManagedBuffer,
         config: OptionalValue<SovereignConfig<Self::Api>>,
-    ) -> ManagedAddress {
+    ) {
         self.tx()
             .to(self.get_chain_factory_address())
             .typed(ChainFactoryContractProxy)
             .deploy_sovereign_chain_config_contract(config)
-            .returns(ReturnsResult)
-            .sync_call()
+            .gas(self.blockchain().get_gas_left() / 2)
```

Comment:
Can you use like this?
```suggestion
            .gas(self.blockchain().get_gas_left())
```

or use a default constant value

Reply:
Then probably is best to have a constant value for deploy